Mathlib/Data/Sym/Basic.lean

```diff
@@ -532,6 +532,29 @@ theorem mem_append_iff {s' : Sym α m} : a ∈ s.append s' ↔ a ∈ s ∨ a ∈
   Multiset.mem_add
 #align sym.mem_append_iff Sym.mem_append_iff
 
+/-- An element of type α can be seen as an element of Sym α 1 --/
+def one (i : α) : Sym α 1 := ⟨[i], by simp only [Multiset.coe_singleton,
+  Multiset.card_singleton]⟩
+
+lemma one_coe (i : α) : (one i : Multiset α) = {i} := rfl
+
+variable (α) in
+lemma one_bijective : Function.Bijective (one : α → Sym α 1) := by
+  refine ⟨fun _ _ h ↦ Multiset.singleton_inj.1 <| by simp [← one_coe, h], fun x ↦ ?_⟩
+  obtain ⟨i, hi⟩ := Multiset.card_eq_one.1 x.2
+  exact ⟨i, ext <| by simp [one_coe, ← hi]⟩
+
+open Multiset Classical in
+@[simps apply]
+noncomputable def one_equiv : α ≃ Sym α 1 where
```

Comment:
If you do this there is no point in keeping `one` (you can call this one `one` if you want). You can apply an equiv to an element, Lean will do its job in using the `toFun` part). Note the tag `@[simps apply]` that produces the lemma
```lean
lemma one_equiv_apply : one_equiv i = ⟨[i], _⟩
```
We surely want this lemma, but `simps` also tag it with `simp`: it's up to you if you want this. In practice the question is "when someone will see `one_equiv i` and they will try `simp`, because we always do that, do we want to see `⟨[i], _⟩` or not?". In practice, is this simpler? In my opinion it seems a good idea, since if `one_equiv i` appears coerced to `Multiset` (something  that I guess is common), `simp` will transform this to `{i}`.

Reply:
Wait, why are you going through lists in the definition?

Reply:
```lean
open Multiset Classical in
@[simps apply]
noncomputable def one_equiv : α ≃ Sym α 1 where
  toFun := fun i ↦ ⟨{i}, by simp only [card_singleton]⟩
  invFun := fun x ↦ choose <| card_eq_one.1 x.2
  left_inv := fun i ↦ by simpa using (choose_spec <| card_eq_one.1
      (⟨[i], by simp only [coe_singleton, card_singleton]⟩ : Sym α 1).2).symm
  right_inv := fun x ↦ by
    simp only [val_eq_coe, coe_singleton]
    exact ext <| by simp_rw [← choose_spec <| card_eq_one.1 x.2]; rfl
```
is more natural, and then my above comment will apply.

Reply:
I'll trust your judgement on this one! So, I just remove `one`, `one_bijective` (`one_coe` too?), and use your latest definition. Do you think it's better to call it `one` or `one_equiv`?

Also, lists? Where am I using lists?

Reply:
Weird issue: if I remove the stuff above `one_equiv`, I get the error `Unexpected token: open`.

Reply:
Even nicer with dot notation
```
open Multiset Classical in
@[simps apply]
noncomputable def one_equiv : α ≃ Sym α 1 where
  toFun := fun i ↦ ⟨{i}, by simp⟩
  invFun := fun x ↦ (card_eq_one.1 x.2).choose
  left_inv := fun i ↦ by simpa using (card_eq_one.1 (⟨{i}, by simp⟩ : Sym α 1).2).choose_spec.symm
  right_inv := fun x ↦ ext <| by simp_rw [← choose_spec <| card_eq_one.1 x.2]; rfl
```

Reply:
`[i]` is a list.

Yes, you can remove completely `one` and `one_coe` (`one_coe` is done by `simp`). I don't understand the error, you probably messed up something. I have the file like this
```lean
theorem mem_append_iff {s' : Sym α m} : a ∈ s.append s' ↔ a ∈ s ∨ a ∈ s' :=
  Multiset.mem_add
#align sym.mem_append_iff Sym.mem_append_iff

open Multiset Classical in
@[simps apply]
noncomputable def one_equiv : α ≃ Sym α 1 where
  toFun := fun i ↦ ⟨{i}, by simp⟩
  invFun := fun x ↦ (card_eq_one.1 x.2).choose
  left_inv := fun i ↦ by simpa using (card_eq_one.1 (⟨{i}, by simp⟩ : Sym α 1).2).choose_spec.symm
  right_inv := fun x ↦ ext <| by simp_rw [← choose_spec <| card_eq_one.1 x.2]; rfl

/-- Fill a term `m : Sym α (n - i)` with `i` copies of `a` to obtain a term of `Sym α n`.
This is a convenience wrapper for `m.append (replicate i a)` that adjusts the term using
`Sym.cast`. -/
def fill (a : α) (i : Fin (n + 1)) (m : Sym α (n - i)) : Sym α n :=
  Sym.cast (Nat.sub_add_cancel i.is_le) (m.append (replicate i a))
#align sym.fill Sym.fill
```
and it works.

Reply:
Concerning the name, I think `one_equiv` is OK, but don't spend too much time in it now.

Reply:
I think it's the comment line above it that generates the error; without the comment, it works fine. I implemented the rest of the suggestions. Thanks again!